drakrun/drakrun/regression.py

```diff
@@ -0,0 +1,140 @@
+import argparse
+import hashlib
+import json
+import os
+import threading
+import tempfile
+import logging
+from dataclasses import dataclass
+from typing import Optional, List
+from pathlib import Path
+
+from karton.core import Karton, Task, Producer, Resource, Config
+from malduck.extractor import ExtractManager, ExtractorModules
+from mwdblib import MWDB
+from drakrun.version import __version__ as DRAKRUN_VERSION
+from contextlib import contextmanager
+
+
+@contextmanager
+def changedLogLevel(logger, level):
+    old_level = logger.level
+    logger.setLevel(level)
+    yield
+    logger.setLevel(old_level)
+
+
+@dataclass
+class TestCase:
+    sha256: str
+    extension: str
+    ripped: str
+    path: Optional[str] = None
+
+    def get_sample(self) -> bytes:
+        # If user provided file path for the sample
+        if self.path:
+            data = open(self.path, "rb").read()
+
+            h = hashlib.sha256()
+            h.update(data)
+            if h.hexdigest() != self.sha256:
```

Comment:
nit: use `.lower()` on both sides to ignore hash text case? otherwise may lead to unintuitive behavior

Reply:
:+1: